src/sidebar/util/test/service-context-test.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const propTypes = require('prop-types');
+const { createElement, render } = require('preact');
+
+const { ServiceContext, withServices } = require('../service-context');
+
+describe('service-context', () => {
+  describe('withServices', () => {
+    let container;
+    let lastProps;
+
+    function TestComponent(props) {
+      lastProps = props;
+    }
+    TestComponent.injectedProps = ['aService'];
+    const WrappedComponent = withServices(TestComponent);
```

Comment:
My assumption that the reason this doesn't throw in the absence of `propTypes` is that this is testing directly against this module and not using the Angular-wrapping/React-controller-creation code? I mean, I can see that this is just a basic object...

Reply:
The code in `withServices` does `Wrapper.propTypes = { ...Component.propTypes }`, if `Component.propTypes` happens to be undefined or null, that just returns an empty object.